cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -175,9 +189,17 @@ public ClassLoader getArtifactClassLoader(ArtifactId artifactId) throws IOExcept
       Field field = Fields.findField(pluginType.getType(), configFieldName);
       TypeToken<?> configFieldType = pluginType.resolveType(field.getGenericType());
       Object config = instantiatorFactory.get(configFieldType).create();
-      Reflections.visit(config, configFieldType.getType(),
-                        new ConfigFieldSetter(pluginClass, plugin.getArtifactId(), plugin.getProperties(),
-                                              getFieldsWithMacro(plugin)));
+
+      // perform macro substitution if an evaluator is provided
+      if (macroEvaluator == null) {
+        Reflections.visit(config, configFieldType.getType(),
+                          new ConfigFieldSetter(pluginClass, plugin.getArtifactId(), plugin.getProperties(),
+                                                getFieldsWithMacro(plugin)));
+      } else {
+        Reflections.visit(config, configFieldType.getType(),
+                          new ConfigFieldSetter(pluginClass, plugin.getArtifactId(),
+                                                substituteMacroFields(plugin, macroEvaluator), new HashSet<String>()));
```

Comment:
@albertshau the reason for passing empty set is, during this plugin instantiation all the macro fields are expanded, so those fields doesn't contain macro anymore.